mlrun/runtimes/function.py

```diff
@@ -1396,3 +1318,130 @@ def get_nuclio_deploy_status(
 
     text = "\n".join(outputs) if outputs else ""
     return state, address, name, last_log_timestamp, text, function_status
+
+
+def set_source_archive(
```

Comment:
Why global function and not a method of the spec ?

Reply:
this is server side logic just like `compile_function_config`, should not be in the client side/sdk (spec)

Reply:
I don't feel like global function is the way to go to differentiate between server and client logic